nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/storage/AzureDataLakeStorageFileResourceService.java

```diff
@@ -0,0 +1,151 @@
+        this.clientFactory = new DataLakeServiceClientFactory(getLogger(), getProxyOptions(context));
+        this.context = context;
+    }
+
+    @OnDisabled
+    public void onDisabled() {
+        this.clientFactory = null;
+        this.context = null;
+    }
+
+    @Override
+    public FileResource getFileResource(Map<String, String> attributes) {
+        DataLakeServiceClient client = getStorageClient(attributes);
+        try {
+            return fetchBlob(client, attributes);
+        } catch (final DataLakeStorageException | IOException e) {
+            throw new ProcessException("Failed to fetch file from ADLS Storage", e);
+        }
+    }
+
+    protected DataLakeServiceClient getStorageClient(Map<String, String> attributes) {
+        final ADLSCredentialsService credentialsService = context.getProperty(ADLS_CREDENTIALS_SERVICE)
+                .asControllerService(ADLSCredentialsService.class);
+        return clientFactory.getStorageClient(credentialsService.getCredentialsDetails(attributes));
+    }
+
+    /**
+     * Fetching file from the provided filesystem and directory in ADLS.
+     *
+     * @param storageClient azure data lake service client
+     * @param attributes configuration attributes
+     * @return fetched file as FileResource
+     * @throws IOException exception caused by missing parameters or blob not found
+     */
+    private FileResource fetchBlob(final DataLakeServiceClient storageClient, final Map<String, String> attributes) throws IOException {
+        final String fileSystem = context.getProperty(FILESYSTEM).evaluateAttributeExpressions(attributes).getValue();
+        final String directory = context.getProperty(DIRECTORY).evaluateAttributeExpressions(attributes).getValue();
+        final String file = context.getProperty(FILE).evaluateAttributeExpressions(attributes).getValue();
+
+        if (isBlank(fileSystem) || isBlank(file)) {
+            throw new ProcessException("Filesystem and file name cannot be empty");
+        }
+        validateDirectoryProperty(directory);
+
+        final DataLakeFileSystemClient fileSystemClient = storageClient.getFileSystemClient(fileSystem);
+        final DataLakeDirectoryClient directoryClient = fileSystemClient.getDirectoryClient(directory);
+        final DataLakeFileClient fileClient = directoryClient.getFileClient(file);
+
+        if (fileClient.getProperties().isDirectory()) {
+            throw new ProcessException(FILE.getDisplayName() + " (" + file + ") points to a directory. Full path: " + fileClient.getFilePath());
+        }
+
+        if (!fileClient.exists()) {
+            throw new ProcessException(String.format("File %s/%s not found in file system: %s", directory, file, fileSystem));
+        }
+
+        return new FileResource(fileClient.openInputStream().getInputStream(),
+                fileClient.getProperties().getFileSize());
+    }
+
+    private static void validateDirectoryProperty(String directoryName) {
```

Comment:
This method can be found in `AbstractAzureDataLakeStorageProcessor`, so it doesn't need to be duplicated.